File: TestApp/app/src/main/java/com/romiope/testapp/LoginActivity.java
package com.romiope.testapp;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.ActionBar;
import android.widget.EditText;
import android.widget.Toast;

import com.romiope.testapp.network.RestClient;
import com.romiope.testapp.network.v1.wrappers.AuthWrapper;

import rx.android.schedulers.AndroidSchedulers;
import rx.schedulers.Schedulers;

public class LoginActivity extends BaseActivity {

    private final AuthWrapper authEndpoint = RestClient.getInstance().getApi().getAuthApi();

    EditText loginInput;
    EditText passInput;

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_login);

        loginInput = findViewById(R.id.et_login);
        passInput = findViewById(R.id.et_password);

        ActionBar actionBar = getSupportActionBar();
        if (actionBar != null) {
            actionBar.hide();
        }

        if (BuildConfig.DEBUG) {
            loginInput.setText("tesonet");
            passInput.setText("partyanimal");
        }
    }

    @Override
    protected void onStart() {
        super.onStart();
        findViewById(R.id.btn_login).setOnClickListener(view -> {
            String login = loginInput.getText().toString();
            String password = passInput.getText().toString();

            subscriptions.add(authEndpoint.performLogin(login, password)
                    .subscribeOn(Schedulers.newThread())
                    .observeOn(AndroidSchedulers.mainThread())
                    .subscribe(tokenResponse -> {
                                    TokenKeeper.setToken(getApplicationContext(), tokenResponse.body().token);
                                    MainActivity.start(getApplicationContext());
                                },
                            throwable -> Toast.makeText(this, throwable.toString(), Toast.LENGTH_LONG).show()));
        });
    }
}

File: Testio/app/src/main/java/lt/marius/testio/LoginActivity.kt
package lt.marius.testio

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.View
import android.view.WindowManager
import kotlinx.android.synthetic.main.activity_login.*

/**
 * A login screen that offers login via email/password.
 */
class LoginActivity : AppCompatActivity() {
	override fun onCreate(savedInstanceState: Bundle?) {
		super.onCreate(savedInstanceState)
		window.setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN,
		                WindowManager.LayoutParams.FLAG_FULLSCREEN)
		setContentView(R.layout.activity_login)


		loginFormLogin.setOnClickListener {
			validateAndRun()
		}

	}

	private fun validateAndRun() {
		loginProgressContainer.visibility = View.VISIBLE
		loginForm.visibility = View.GONE
		loginProgressMessage.setText(R.string.login_progress_message_login)
	}
}

